koku/api/settings/settings.py

```diff
@@ -247,37 +247,37 @@ def _tag_key_handler(self, settings):
-            enabled_tags_no_abbr = []
-            tag_view = obtainTagKeysProvidersParams[provider_name]["tag_view"]
-            query_handler = obtainTagKeysProvidersParams[provider_name]["query_handler"]
-            enabled_tag_keys = obtainTagKeysProvidersParams[provider_name]["enabled_tag_keys"]
-            provider = obtainTagKeysProvidersParams[provider_name]["provider"]
-            available, _ = self._obtain_tag_keys(tag_view, query_handler, enabled_tag_keys)
-
-            # build a list of enabled tags for a given provider, removing the provider name prefix
-            for enabled_tag in settings.get("enabled", []):
-                if enabled_tag.startswith(provider_name + tag_delimiter):
-                    enabled_tags_no_abbr.append(enabled_tag.split(tag_delimiter, 1)[1])
-
-            invalid_keys = [tag_key for tag_key in enabled_tags_no_abbr if tag_key not in available]
-
-            if invalid_keys:
-                key = "settings"
-                message = f"Invalid tag keys provided: {', '.join(invalid_keys)}."
-                raise ValidationError(error_obj(key, message))
-
-            if "aws" in provider_name:
-                existing_enabled_tags = list(
-                    enabled_tag_keys.objects.filter(enabled=True).values_list("key", flat=True)
-                )
-
-                if enabled_tags_no_abbr != existing_enabled_tags:
-                    updated[ix] = update_enabled_keys(self.schema, enabled_tag_keys, enabled_tags_no_abbr)
-
-            else:
-                remove_tags = []
-                with schema_context(self.schema):
+        with schema_context(self.schema):
+            for ix, provider_name in enumerate(obtainTagKeysProvidersParams):
+                enabled_tags_no_abbr = []
+                tag_view = obtainTagKeysProvidersParams[provider_name]["tag_view"]
+                query_handler = obtainTagKeysProvidersParams[provider_name]["query_handler"]
+                enabled_tag_keys = obtainTagKeysProvidersParams[provider_name]["enabled_tag_keys"]
+                provider = obtainTagKeysProvidersParams[provider_name]["provider"]
+                available, _ = self._obtain_tag_keys(tag_view, query_handler, enabled_tag_keys)
+
+                # build a list of enabled tags for a given provider, removing the provider name prefix
+                for enabled_tag in settings.get("enabled", []):
+                    if enabled_tag.startswith(provider_name + tag_delimiter):
+                        enabled_tags_no_abbr.append(enabled_tag.split(tag_delimiter, 1)[1])
+
+                invalid_keys = [tag_key for tag_key in enabled_tags_no_abbr if tag_key not in available]
+
+                if invalid_keys:
+                    key = "settings"
+                    message = f"Invalid tag keys provided: {', '.join(invalid_keys)}."
+                    raise ValidationError(error_obj(key, message))
+
+                if provider_name in ("aws", "azure"):
```

Comment:
Should we do something like this instead here? `if provider_name in (Provider.PROVIDER_AZURE, Provider.PROVIDER_AWS):`

Reply:
Nope. It's referring to a dict defined at the top of the file.